### DISCUSSION TITLE
Search Filter Tutorial isn't working on my data

### DISCUSSION THREAD

**Discussion by bamboowonder** [Budibase/budibase#5350] [Help]
I'm trying to follow the the tutorial video here: 
https://www.youtube.com/watch?v=hmmLdeY-5d4&t=139s
https://docs.budibase.com/docs/searchfilter-data

But at the step where the "select" is supposed to display values from the schema, the Select isn't displaying anything.   
I've set the parent form to the schema = database table i want to search
and the Select Field is set to the column I want to search on.   And the OptionSource is "schema"

Could it be because that column has nulls?  Or that it contains 30,000 records? 
The column referenced only has about 30 unique values.

I matched everything I'm doing step by step to the video (as much as possible as video is using an year-old version)

User facing search is absolutely critical for me as we need more sophistication than the Dynamic search can provide.
eg:   OR in addition to AND.
where column ilike any('{%value1%,%value2%}')   so that we can ILIKE search on an array of values  (the current IN only allows for exact values, and the ILIKE doesn't allow for multiple values)

**Comment by melohagan** [comment]
Hey @bamboowonder 

Would you mind sharing your app export?

**Reply by bamboowonder** [reply]
Here is a demo app.
[testapp-export-1649699284849.txt](https://github.com/Budibase/budibase/files/8466563/testapp-export-1649699284849.txt)

**Comment by melohagan** [comment]
Whenever you use schema as the option source, then the column must be 'Options' type, with values provided:

![Screenshot 2022-04-12 at 10 06 50](https://user-images.githubusercontent.com/101575380/162924105-3dbf9640-89e7-43d9-8515-b06cf7ff45c6.png)

This note was missing from the docs, but I have updated them here: https://docs.budibase.com/docs/option-picker

Thanks for your post!

**Reply by bamboowonder** [reply]
From watching your youtube video it seems like it was grabbing option values from db.  Is the Select Options able to be generated by looking for distinct values from the column to be searched?

**Reply by melohagan** [reply]
Yes you can accomplish that by using a data provider as the options source.

**Comment by bamboowonder** [comment]
Somewhat related, it doesn't look like I can easily swap out the Select with a Multiselect.  Is that because it is expecting an array?  
Is there are recommended format that the Multiselect is expecting when using a dataprovider source?
And then for binding the Multiselect to a table, I'm guessing I'd have to write a custom select that accepts an array in the WHERE?

I'm asking in case you already have a best practice for dealing with using Multiselect for filtering a table.

Thanks a bunch

**Reply by melohagan** [reply]
Hi @bamboowonder 

There is a separate picker for multi-select: https://docs.budibase.com/docs/multi-select-picker

The options get listed in the same way as the Options type, but in this case the type is 'Multi-select'

![Screenshot 2022-04-14 at 09 04 16](https://user-images.githubusercontent.com/101575380/163341585-550fffe2-e8fd-4ea8-81a3-a2a04ba88b80.png)

**Reply by bamboowonder** [reply]
I know.  But the multiselect does not like using a dataprovider column as it's source the way the select does.
My question was if I need to do something different with it?

**Reply by melohagan** [reply]
I get you now, and it does appear to have some limitations. There is a related issue opened here: https://github.com/Budibase/budibase/issues/3773
